tests/acceptance/features/bootstrap/TagsContext.php

```diff
@@ -1627,36 +1627,41 @@ public function theAdministratorRemovesTheTagFromFileSharedByUsingTheWebdavApi(
 	}
 
 	/**
-	 * @When user :user searches for tag :tagName using the webDAV API
+	 * search resources with tags using the REPORT webDAV method
 	 *
-	 * @param {string} $user
-	 * @param {string} $tagName
+	 * @param $user
+	 * @param TableNode $tagNames
 	 *
 	 * @return void
+	 * @throws Exception
 	 */
-	public function userSearchesForTagUsingWebDavAPI($user, $tagName) {
+	public function searchForTagsOfFileWithReportUsingWebDAVApi($user, $tagNames) {
+		$this->featureContext->verifyTableNodeColumnsCount($tagNames, 1);
+		$tagNames = $tagNames->getRows();
 		$baseUrl = $this->featureContext->getBaseUrl();
 		$password = $this->featureContext->getPasswordForUser($user);
 		$createdTagsArray = $this->getListOfCreatedTags();
+		$createdTagIds = [];
+		$createdTagNames = [];
+		foreach ($createdTagsArray as $tagId => $tagArray) {
+			\array_push($createdTagIds, $tagId);
+			\array_push($createdTagNames, $tagArray['name']);
+		}
 		$body = "<?xml version='1.0' encoding='utf-8' ?>\n" .
 			"	<oc:filter-files xmlns:d='DAV:' xmlns:oc='http://owncloud.org/ns' >\n" .
 			"		<oc:filter-rules>\n";
-		$tagIds = [];
-		$tagNames = [];
-		foreach ($createdTagsArray as $tagId => $tagArray) {
-			\array_push($tagIds, $tagId);
-			\array_push($tagNames, $tagArray['name']);
-		}
-		$found = \in_array($tagName, $tagNames);
-		if ($found) {
-			$index = \array_search($tagName, $tagNames);
-			$body .=
-				"			<oc:systemtag>$tagIds[$index]</oc:systemtag>\n";
-		} else {
-			throw new Error(
-				"Expected: Tag with name $tagName to be in created list, but not found!" .
-				"List of created Tags: " . \implode(",", $tagNames)
-			);
+		foreach ($tagNames as $tagName) {
+			$found = \in_array($tagName[0], $createdTagNames);
+			if ($found) {
+				$index = \array_search($tagName[0], $createdTagNames);
+				$body .=
+					"			<oc:systemtag>$createdTagIds[$index]</oc:systemtag>\n";
+			} else {
+				throw new Error(
+					"Expected: Tag with name $tagName[0] to be in created list, but not found!" .
+					"List of created Tags: " . \implode(",", $tagNames)
```

Comment:
```suggestion
					"List of created Tags: " . \implode(",", $createdTagNames)
```
Maybe it should be like this now?